Keter/Types/Middleware.hs

```diff
@@ -26,7 +27,31 @@ import Data.ByteString  as S (ByteString)
 import Data.Text.Lazy.Encoding as TL (encodeUtf8, decodeUtf8)
 import Data.Text.Encoding as T (encodeUtf8, decodeUtf8)
 import Data.String (fromString)
-import qualified Data.HashMap.Strict as H
+
+#if MIN_VERSION_aeson (2,0,0)
+import qualified Data.Aeson.Key as K
+import qualified Data.Aeson.KeyMap as M
+import qualified Data.Text as Text
+import qualified Data.ByteString.Char8 as C8 (unpack)
+import qualified Data.ByteString.Lazy.Char8 as CL8 (unpack)
+
+
+toKey :: Text.Text -> K.Key
+toKey = K.fromText
+
+fromKey :: K.Key -> Text.Text
+fromKey = K.toText
+
+#else
+import qualified Data.HashMap.Strict as M
+
+toKey :: Text.Text -> Text.Text
+toKey = id
+
+fromKey :: Text.Text -> Text.Text
+fromKey = id
+
+#endif
```

Comment:
I think it'd be better to move these functions in a separate module, because you've also defined toKey in FilePath,
that would also avoid introducing CPP in both midleware and filepath